packages/gestalt/src/commonTypes.js

```diff
@@ -24,6 +21,9 @@ export type DismissButtonType = {|
   onDismiss: () => void,
 |};
 
+const OnNavigationOptionsPropType: React$PropType$Primitive<OnNavigationOptionsType> =
```

Comment:
I don't think we want to define this type here; it makes much more sense in `onNavigation.js`